src/Waives.Http/ReliableRequestSender.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Polly;
+using Polly.Extensions.Http;
+using Polly.Retry;
+
+namespace Waives.Http
+{
+    internal class ReliableRequestSender : IHttpRequestSender
+    {
+        private readonly IHttpRequestSender _wrappedRequestSender;
+        private readonly RetryPolicy<HttpResponseMessage> _policy;
+
+        public ReliableRequestSender(Action<DelegateResult<HttpResponseMessage>, TimeSpan, int, Context> retryAction, IHttpRequestSender wrappedRequestSender)
+        {
+            var sleepDurationProvider = new ExponentialBackoffSleepProvider();
+
+            _policy = HttpPolicyExtensions
+                .HandleTransientHttpError()
```

Comment:
💪 Do we ever see HTTP errors, or are they always wrapped up in `WaivesApiException`s as per the `ExceptionHandlingRequestSender`?

Reply:
No, they are always wrapped up. The `HandleTransientHttpError` actually covers both HTTP errors and status code errors. But you've prompted me to change this to:
```
_policy = Policy
                .Handle<WaivesApiException>()
                .OrTransientHttpStatusCode()
```
which is actually the subset of behaviour we need